### ISSUE TITLE
Conflict with `gatsby-transformer-json`

### ISSUE THREAD

**Issue by crazko** [thinhle-agilityio/gatsby-source-apiserver#38]
Hi, I get an error when my project depends on both this plugin and `gatsby-transformer-json`. The error is thrown during the build process:

```
"gatsby-transformer-json" threw an error while running the onCreateNode lifecycle:

Could not find function loadNodeContent for plugin gatsby-source-apiserver
```

While development mode still works, it fails during the production build and never run  till the end.

I reproduced this error in https://codesandbox.io/s/sad-wilbur-xwugm - just restart the sandbox to see errors in the terminal.

Thank you for any ideas!

**Comment by crazko** [comment]
I've have naively tried a workaround mentioned in https://github.com/ldd/gatsby-source-github-api/issues/10#issuecomment-525575781 locally - switch node `mediaType` from `application/json` to `text/plain`.

- development mode **works** without any arror
- production mode **fails** on "Building static HTML for pages":

![Annotation 2020-01-20 190207](https://user-images.githubusercontent.com/1255989/72748384-62bf9c80-3bb7-11ea-8aba-598655900b4d.png)

**Comment by ff4500** [comment]
I'm running into a similar issue on my build, and tried the fix posted above as well.

I went ahead and tried changing line in [normalize.js](https://github.com/thinhle-agilityio/gatsby-source-apiserver/blob/aec30e65c32cf0b02f052469377f85e0af05eea7/src/normalize.js#L76) from `mediaType: 'application/json',` to `mediaType: 'text/plain',` and my build is working again locally with no errors.

It sure looks like since the [gatsby-transformer-json](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-json/src/gatsby-node.js#L40-L64) is looking for the `application/json` media type and it sees it, it's choking on the `loadNodeContent` field not being there. If the mediaType is set as something else (ie: `text/plain`), it just ignores the node(s) that the apiserver is creating. Since I can't patch my automated build, it's failing on a production build on Netlify right now.

Any thoughts on a better workaround?

**Comment by alexej-d** [comment]
Any permanent solutions to this?

**Comment by Pearce-Ropion** [comment]
This started happening to me immediately after updating to 2.1.7 (from 2.1.3). I didn't change my config at all.

**Comment by mAAdhaTTah** [comment]
I'm running into this. I used @ff4500's solution with [`patch-package`](https://www.npmjs.com/package/patch-package). I would love to upstream this but probs won't have time for a few weeks.

**Comment by Pearce-Ropion** [comment]
I just updated all of my gatsby packages to the latest versions but I am still unable to update this one with the same error. Not sure if there is any progress on this.

**Comment by joernroeder** [comment]
also running into this now. any update or fix?

**Comment by mmorsel** [comment]
I had same problem with gatsby-source-apiserver, but i find alternative plugin: [gatsby-source-custom-api](https://www.gatsbyjs.com/plugins/gatsby-source-custom-api/). Now i hav`t conflict with gatsby-transformer-json.

**Comment by Pearce-Ropion** [comment]
I got a fix for this working. See #64 
Not sure how active the plugin author is so feel free to use my pre-built fork in your repos until it gets merged.
https://github.com/Pearce-Ropion/gatsby-source-apiserver/tree/fork/transform-json-error
```
npm install Pearce-Ropion/gatsby-source-apiserver#6112115 -D 
yarn add Pearce-Ropion/gatsby-source-apiserver#6112115 -D 
```

**Comment by max-md** [comment]
Having the same issue, have you hear anything back from the author @Pearce-Ropion ?

**Comment by Pearce-Ropion** [comment]
> Having the same issue, have you hear anything back from the author @Pearce-Ropion ? 

No. Nothing yet. You can use my custom fork, or version 2.1.3. I ended up making a custom source plug-in for my repo that replicate ssome of this behavior as well adds new functionality￼ due to this plugin not having enough error handling.

**Comment by max-md** [comment]
@Pearce-Ropion Oh thank you so much. I didn't realize that version worked. I will use that for now, thanks. But I hope you get your PR merged soon 🤞
